### DISCUSSION TITLE
Carrier duty cycle issues for JVC devices

### DISCUSSION THREAD

**Discussion by eismeraldo** [Arduino-IRremote/Arduino-IRremote#1191] [Help] [answered]
Hello, I have an L336 learning remote control and a partially defective JVC remote control. I was able to read almost all button codes using the Arduino program “ReceiveDemo”.
The L336 can be easily programmed with the JVC remote control, but not with the Arduino program, based on the example: SimpleSender
Now I think I have found the problem. According to the documentation from JVC "RemoteCodes.pdf", the carrier frequency has a duty cycle of approx. 33% and not 50%. Specifically, a period of 26.4us is divided into 8.8us high and 17.6us low.
Question: Can I configure something so that the duty cycle meets JVC requirements?
[RemoteCodes.pdf](https://github.com/Arduino-IRremote/Arduino-IRremote/files/13539890/RemoteCodes.pdf)
![carrier frequency_JVC_Confidential_PDF](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/3942895/de1da98b-e308-486c-9e55-36782f965ac7)
![carrier frequency_JVC_Remotecontrol](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/3942895/8c72ed34-8341-4747-b0c2-f4d2116739a8)
JVC Remote Control
![carrier frequency_Arduino](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/3942895/26a917e0-5282-4974-bb2d-6a93a80ea6df)
Arduino

**Comment by AnalysIR** [comment]
unlikely related to duty cycle.....maybe post your IR send circuit

**Reply by eismeraldo** [reply]
The diagram in the red frame is the transmitter. I used the two receiver circuits to check the functionality of the Arduino transmitter and the JVC remote control. The first receiver with the diode shows the carrier frequency and the second receiver shows the signal. The second Arduino then shows the address and code that match the transmitter and the JVC remote control.
![Test_IR_Sender_Empfänger_Schema2](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/3942895/bbd1658e-4e69-4bf2-b2d3-714c097adb10)

**Comment by AnalysIR** [comment]
you are getting less than 20mA current thru the IR LED.

Try changing R3 from 180R to 22R or 10R or the closest you have.

Also make sure you place the emitter as close as possible to the learning remote when capturing..

**Reply by eismeraldo** [reply]
Thanks for your answer. This doesn't help, it's the problem with the duty cycle, which won't be changed. As I said, the transmitter and receiver hardware works, I tested it in a variety of ways. With this constellation I was also able to read the codes of the JVC remote control and test whether the transmitter was working "correctly".

**Reply by ArminJo** [reply, marked as answer]
The output is generated with a [30% duty cycle](https://github.com/Arduino-IRremote/Arduino-IRremote/blob/master/src/IRSend.hpp#L1192) . 
How do you manage to get a 50% one?

**Reply by AnalysIR** [reply]
Actually it looks more than 50%.....

**Reply by eismeraldo** [reply]
I'm so sorry for taking up your time unnecessarily. I apologize to you from the bottom of my heart.
As you wrote, the carrier frequency from the Arduino is correct, it has a duty cycle of 30%. However, there are fluctuations, but it shouldn't be too problematic.
![carrier frequency_BC338_01](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/3942895/ad872f50-29dd-459a-8730-3493785d5659)
The transistor BC338 causes the delay of 2.5us. 
![carrier frequency_BC338_02](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/3942895/794b9a71-4990-49ee-aa75-d5a0e3204ef7)
![carrier frequency_BC338_03](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/3942895/f47c7c51-4aad-4b8d-9292-0466aec6b548)
I replaced it with a MOSFET 2N7000.
![carrier frequency_MOSFET_2N7000_00_Schema](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/3942895/cc7223bc-d644-48e9-9008-b4912cb1bb8a)
Now the carrier frequency has a duty cycle of 30%.
![carrier frequency_MOSFET_2N7000_01](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/3942895/d453db2c-d1ad-49c7-96e1-953ad0e88dd2)
![carrier frequency_MOSFET_2N7000_02](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/3942895/ee8cc5ea-7915-47a5-9b7f-a1cf99bf934b)
The programmable remote control accepts the address and code and the radio can be controlled again via the new remote control.
Thank you again for your help and commitment.
